From 563409ed82164663e08835ed07584337ce601773 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Mon, 8 Jun 2015 07:27:23 +0000
Subject: [PATCH] abuild: clear inherit provides and install_if for subpackages

Unexpected result will happen if subpackages inherits the provides or
install_if from main package. So we clear them.
---
 abuild.in | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/abuild.in b/abuild.in
index 9e6f43a..cb069f2 100644
--- a/abuild.in
+++ b/abuild.in
@@ -947,6 +947,11 @@ archcheck() {
 
 prepare_package() {
 	msg "Preparing ${subpkgname:+sub}package ${subpkgname:-$pkgname}..."
+	if [ -n "$subpkgname" ]; then
+		# the subpackages should not inherit those
+		provides=""
+		install_if=""
+	fi
 	stripbin
 	prepare_metafiles \
 		&& prepare_trace_rpaths \
-- 
2.4.2

